### DISCUSSION TITLE
Prevent new billing entry from being created if checks fail

### DISCUSSION THREAD

**Discussion by RolandvonBorstel** [sadieblue/agilityblue.workspace-scripts#17] [Q&A]
I have created a script with a on create/before save trigger on billing entries. It does return $null if certain criteria are not met and $InboundEntry otherwise. Unfortunately, it does not stop a new entry from being created if $null is returned.

Is there another way to prevent the new entry being created?

**Comment by bmward80** [comment]
Hi @RolandvonBorstel,

The behavior should be that if the same type is not passed back, the user should receive a notice that saving was not possible due to a script blocking it. If that's not the case, can you provide me with an example to recreate this issue?

Brandon

**Reply by bmward80** [reply]
I just tried this scenario with a very simple before save setup on a billing entry:

```powershell
$inboundEntry = Get-InboundObjectInstance

if ($inboundEntry.Quantity -eq 1.00) {
    return $null
}

return $inboundEntry
```

When I tried to save my billing entry with a quantity of 1, I would get the error and it would not save, and when I would give any other quantity, it would save. So, now I am curious how your script/example is setup that would be causing it to not have the expected behavior.

Brandon

**Reply by RolandvonBorstel** [reply]
Hi Brandon,

The test script has one single line:  `return $null`

The trigger on it is: 
![image](https://github.com/user-attachments/assets/7bbfd72c-dcf3-4223-b08f-83f7a4fafd33)

I would assume this script will prevent any new billing entry from being created as it will always return $null. Unfortunately, the UI simply saves the new entry without any error message.

**Reply by bmward80** [reply]
Yeah, I definitely get a blocked error when trying just a one liner `return $null`

![image](https://github.com/user-attachments/assets/1a2f6c3a-4544-499d-849b-560adb54c451)

Hmm. Do you have any other triggers associated with billing entries, such as an after save trigger?

Brandon

**Reply by RolandvonBorstel** [reply]
We may have other scripts with on create/after save triggers. Not sure if it matters that this happens in our sandbox workspace.

**Reply by bmward80** [reply]
Allright, I'll open a ticket at our support email for this to continue the discussion over there because it looks like we may need to dig further into your specific workspace details to determine what's causing the behavior and we'd want to discuss that privately.

Brandon

**Reply by RolandvonBorstel** [reply]
I have now tested the same script in the production workspace and I do get the error. It must be something weird in the sandbox.
